jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -356,19 +418,51 @@ def run(self):
 
         search_params = search_params_parser(self.options.pop("search_params", False))
 
+        new_tab = self.options.pop("new_tab", False)
+
+        content = self.content
+
+        button_text = None
+
         prefix = os.path.relpath(
             os.path.join(self.env.app.srcdir, JUPYTERLITE_DIR),
             os.path.dirname(self.get_source_info()[0]),
         )
 
+        if new_tab:
+            directive_button_text = self.options.pop("button_text", None)
+            if directive_button_text is not None:
+                button_text = directive_button_text
+            else:
+                button_text = self.env.config.replite_button_text
```

Comment:
```suggestion
        if new_tab:
            directive_button_text = self.options.pop("button_text", None)
            if directive_button_text is not None:
                button_text = directive_button_text
            else:
                button_text = self.env.config.replite_button_text
            return [
                RepliteTab(
                    prefix=prefix,
                    width=width,
                    height=height,
                    prompt=prompt,
                    prompt_color=prompt_color,
                    content=content,
                    search_params=search_params,
                    lite_options=self.options,
                    button_text=button_text,
                )
            ]
```

It makes sense to put this here to avoid having to branch on `new_tab` again.

Reply:
Thanks – addressed in 2f96d85b6d259563ec6118dc0808f7ec234f356e.